dask/array/creation.py

```diff
@@ -314,10 +314,12 @@ def meshgrid(*xi, **kwargs):
 
     xi = [e.flatten() for e in xi]
     dimensions = [xi[i].size for i in range(len(xi))]
+    chunks = [xi[i].chunks[0] for i in range(len(xi))]
```

Comment:
It would be good to coerce all arguments to dask arrays first, using `asarray()`.

Otherwise, this will error when one of the arguments is a numpy array. (It would be good to add test coverage, too.)